clc/config/config.go

```diff
@@ -0,0 +1,240 @@
+			if err := sc.SetCAPath(cp); err != nil {
+				return cfg, err
+			}
+		}
+		cp := props.GetString(clc.PropertySSLCertPath)
+		kp := props.GetString(clc.PropertySSLKeyPath)
+		kps := props.GetString(clc.PropertySSLKeyPassword)
+		if cp != "" && kp != "" {
+			cp = paths.Join(wd, cp)
+			lg.Debugf("Certificate path: %s", cp)
+			kp = paths.Join(wd, kp)
+			lg.Debugf("Key path: %s", kp)
+			if kps != "" {
+				lg.Debugf("Key password: XXX")
+				if err := sc.AddClientCertAndEncryptedKeyPath(cp, kp, kps); err != nil {
+					return cfg, err
+				}
+			} else {
+				if err := sc.AddClientCertAndKeyPath(cp, kp); err != nil {
+					return cfg, err
+				}
+			}
+		}
+	}
+	cfg.ClientName = makeClientName()
+	usr := props.GetString(clc.PropertyClusterUser)
+	pass := props.GetString(clc.PropertyClusterPassword)
+	if usr != "" || pass != "" {
+		lg.Debugf("Cluster user: %s", usr)
+		lg.Debugf("Cluster password: XXX")
+		cfg.Cluster.Security.Credentials.Username = usr
+		cfg.Cluster.Security.Credentials.Password = pass
+	}
+	return cfg, nil
+}
+
+func makeClientName() string {
+	cn := os.Getenv(envClientName)
+	if cn != "" {
+		return cn
+	}
+	var userName string
+	u, err := user.Current()
+	if err != nil {
+		userName = "UNKNOWN"
+	} else {
+		userName = u.Username
+	}
+	var hostName string
+	host, err := os.Hostname()
+	if err != nil {
+		host = "UNKNOWN"
+	} else {
+		hostName = host
+	}
+	t := time.Now().Unix()
+	return fmt.Sprintf("%s@%s-%d", userName, hostName, t)
+}
+
+// DirAndFile returns the configuration directory and file separately
+func DirAndFile(path string) (string, string, error) {
+	path = filepath.ToSlash(path)
+	// easy case, path is just a config name
+	if strings.Index(path, "/") < 0 {
+		return paths.ResolveConfigDir(path), paths.DefaultConfig, nil
+	}
+	fi, err := os.Stat(path)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return "", "", err
+	}
+	// if path exists, return early
+	if !errors.Is(err, os.ErrNotExist) {
+		if fi.IsDir() {
+			return path, paths.DefaultConfig, nil
+		}
+		// path is a directory, split and send
+		d, f := filepath.Split(path)
+		return strings.TrimSuffix(d, "/"), f, err
+	}
+	// if path doesn't exist, check whether it's file like
+	ext := filepath.Ext(path)
+	if ext == "" {
+		// this is probably a directory
+		return path, paths.DefaultConfig, nil
+	}
+	// this is a file
+	d, f := filepath.Split(path)
+	return strings.TrimSuffix(d, "/"), f, nil
+}
+
+func CreateYAML(opts clc.KeyValues[string, string]) string {
+	// TODO: refactor this function to be more robust, probably using Viper
+	sb := &strings.Builder{}
+	copySection("", 0, sb, opts)
+	return sb.String()
+}
+
+func copySection(name string, level int, sb *strings.Builder, opts clc.KeyValues[string, string]) {
+	if len(opts) == 0 {
+		return
+	}
+	var leaves clc.KeyValues[string, string]
+	var sect clc.KeyValues[string, string]
+	sub := map[string]clc.KeyValues[string, string]{}
+	for _, opt := range opts {
```

Comment:
Can we check for possible options? 
For example I can add config with unrelated configuration options.
For example;

```
config add aliveli/pr-621 aaa=bbb ccc=dddd
aaa: bbb
ccc: dddd
```

Reply:
And also when I try the following command;

```
config add aliveli/pr-621 aaa.bbb.ccc.ddd='bbb' ccc.bbb='dddd'
Created yaml is;
aaa:
  bbb.ccc.ddd:
    ccc.ddd:
      ddd: bbb
ccc:
  bbb: dddd

It should be;
aaa:
  bbb:
    ccc:
      ddd: bbb
ccc:
  bbb: dddd
```

Reply:
It's OK to add any option, since a plugins may use their own configuration options.